app/models/manageiq/providers/vmware/infra_manager/provision_workflow.rb

```diff
@@ -144,6 +144,16 @@ def available_vlans_and_hosts(options = {})
     return vlans, hosts
   end
 
+  def filter_hosts_by_vlan_name(all_hosts)
+    vlan_name = get_value(@values[:vlan])
+    return all_hosts unless vlan_name
+
+    _log.info("Filtering hosts with the following network: <#{vlan_name}>")
+    shared = !vlan_name.match(/dvs_/).nil?
+    vlan_name.sub!(/^dvs_/, '') if shared
+    all_hosts.select { |h| h.lans.any? { |lan| lan.name == vlan_name && !lan.switch.shared.blank? == shared } }
```

Comment:
is this doing n+1 queries possibly (n+1)^2 ? Or all_hosts already has ```all_hosts.include(:lans => switch).references(:lans => switch)``` #haven't tested the query